cpp/src/aztec/honk/sumcheck/polynomials/barycentric_data.test.cpp

```diff
@@ -0,0 +1,42 @@
+#include "barycentric_data.hpp"
+#include <ecc/curves/bn254/fr.hpp>
+
+#include <gtest/gtest.h>
+#include <numeric/random/engine.hpp>
+
+using namespace honk::sumcheck;
+namespace test_sumcheck_polynomials {
+
+template <class FF> class BarycentricDataTests : public testing::Test {};
+
+using FieldTypes = testing::Types<barretenberg::fr>;
+TYPED_TEST_SUITE(BarycentricDataTests, FieldTypes);
+
+#define BARYCENTIC_DATA_TESTS_TYPE_ALIASES using FF = TypeParam;
```

Comment:
There are tests along these lines in univariate.test.cpp also. Do you mind unifying the tests, either moving some of those here, or moving these tests into that file?